File: 백준/Gold/1759. 암호 만들기/암호 만들기.py

L, C = map(int, input().split())

arr = []
arr = input().split()

mo = "aeiou"
mo = list(mo)

sol = [0 for _ in range(L)]
arr.sort()


def back(a, b):
    count = 0
    if a == L:
        for i in range(L):
            for k in mo:
                if sol[i] == k:
                    count += 1
        if count > 0 and L-count > 1:
            for idx in sol:
                print(idx, end='')
            print()
        return
    elif b >= C:
        return

    for j in range(b, C):
        sol[a] = arr[j]
        back(a+1, j+1)


back(0, 0)

File: 백준/Gold/1759. 암호 만들기/암호 만들기.cc
#include<iostream>
#include<algorithm>
#include<string>

using namespace std;

int L, C;
char arr[16] = { 0, };
char sol[16] = { 0, };
string mo = "aeiou";


void dfs(int idx, int count) {

	if (count == L) {
		int mo_n = 0;
		int ja_n = 0;
		
		for (int i = 1; i <= L; i++) {
			for (int j = 0; j < mo.size(); j++) {
				if (sol[i] == mo[j]) {
					mo_n++;
					break;
				}
			}
		}

		ja_n = L - mo_n;

		if (mo_n >= 1 && ja_n>= 2) {
			for (int k = 1; k <= L; k++) {
				cout << sol[k];
			}
			cout << '\n';
		}
		return;
	}

	for (int i = idx; i <= C; i++) {
		sol[count+1] = arr[i];
		dfs(i+1, count + 1);
	}
}

int main() {

	cin >> L >> C;

	for (int i = 1; i <= C; i++) {
		cin >> arr[i];
	}

	sort(arr + 1, arr + 1 + C);

	dfs(1,0);


	return 0;
}